### PR TITLE
cmd/contour: Stub out --contour-config-name flag

### PR DESCRIPTION
**Pull request by stevesloka**
Stubs out the '--contour-config-name' flag on 'contour serve' command
which will take precendence over any other configuration items if
specified.

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

File: site/content/docs/main/configuration.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Many of these flags are mirrored in the [Contour Configuration File](#configurat
 | Flag Name         | Description        |
 |-------------------|--------------------|
 | `--config-path`       | Path to base configuration |
+| `--contour-config-name`       | Name of the ContourConfiguration file to use |
```
Comment (sunjayBhatia) [inline site/content/docs/main/configuration.md:22]:
```suggestion
| `--contour-config-name`       | Name of the ContourConfiguration resource to use |
```

File: cmd/contour/serve.go [hunk 2]
```diff
@@ -212,6 +220,29 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		log.WithError(err).Fatal("unable to create scheme")
 	}
 
+	// Get the ContourConfiguration CRD if specified
+	if len(ctx.contourConfigurationName) > 0 {
+
+		// Determine the name/namespace of the configuration file utilizing the environment
```
Comment (sunjayBhatia) [inline cmd/contour/serve.go:226]:
```suggestion
		// Determine the name/namespace of the configuration resource utilizing the environment
```

File: cmd/contour/serve.go [hunk 3]
```diff
@@ -75,7 +75,8 @@ func registerServe(app *kingpin.Application) (*kingpin.CmdClause, *serveContext)
 	serve := app.Command("serve", "Serve xDS API traffic.")
 
 	// The precedence of configuration for contour serve is as follows:
-	// config file, overridden by env vars, overridden by cli flags.
+	// If ContourConfiguration file is specified, it takes precedence,
```
Comment (sunjayBhatia) [inline cmd/contour/serve.go:78]:
```suggestion
	// If ContourConfiguration resource is specified, it takes precedence,
```

**Review by youngnick — APPROVED** [review]
LGTM, good idea to get this in now.

**Review by sunjayBhatia — APPROVED** [review]

**Review by sunjayBhatia — CHANGES_REQUESTED** [review]
do we still need RBAC additions for the contour config resource?

**Comment by stevesloka** [comment]
> do we still need RBAC additions for the contour config resource?

Yup we do along with the informer bits, etc. I can add that to this PR which probably makes sense.

File: internal/k8s/informers.go [hunk 4]
```diff
@@ -31,12 +31,19 @@ import (
 // +kubebuilder:rbac:groups="projectcontour.io",resources=extensionservices,verbs=get;list;watch
 // +kubebuilder:rbac:groups="projectcontour.io",resources=extensionservices/status,verbs=create;get;update
 
+// +kubebuilder:rbac:groups="projectcontour.io",resources=contourconfigurations,verbs=get;list;watch
+// +kubebuilder:rbac:groups="projectcontour.io",resources=contourconfigurations/status,verbs=create;get;update
+// +kubebuilder:rbac:groups="projectcontour.io",resources=contourdeployments,verbs=get;list;watch
```
Comment (sunjayBhatia) [inline internal/k8s/informers.go:36]: now that i think of it, is access to the `contourdeployment` resource needed in the Contour controller? i *think* only the operator etc. will read/update status on it?

Comment (stevesloka) [reply internal/k8s/informers.go:36]: I thought we'd retire the operator repo and add a new controller to Contour to manage the Deployment bits, but I do agree that at the moment, it's not being used.

Comment (youngnick) [reply internal/k8s/informers.go:36]: Yeah, I think that we should keep this stuff isolated until we start bringing the operator into this repo, so better to leave out for now.

Comment (stevesloka) [reply internal/k8s/informers.go]: Yup makes sense! I took the `ContourDeployment` out for now.

**Review by skriss — APPROVED** [review]
LGTM

**Review by sunjayBhatia — APPROVED** [review]

**Pull request merged**